File: Sample Code/ObjcSampleCode/Demo/DJISdkDemo/Demo/FlightController/General/FCGeneralControlViewController.m
//
//  FCGeneralControlViewController.m
//  DJISdkDemo
//
//  Copyright © 2015 DJI. All rights reserved.
//
/**
 *  DJIFlightController provides different sets of methods to control the movement of the aircraft. This file demonstrates the basic methods
 *  to make the aircraft take-off, go home and land. For more advanced methods, please refer to FCVirtualStickViewController.m.
 */
#import "FCGeneralControlViewController.h"
#import "DemoComponentHelper.h"
#import "DemoAlertView.h"
#import <DJISDK/DJISDK.h>

@interface FCGeneralControlViewController ()
- (IBAction)onTakeoffButtonClicked:(id)sender;
- (IBAction)onGoHomeButtonClicked:(id)sender;
- (IBAction)onLandButtonClicked:(id)sender;

@end

@implementation FCGeneralControlViewController

- (IBAction)onTakeoffButtonClicked:(id)sender {
    DJIFlightController* fc = [DemoComponentHelper fetchFlightController];
    if (fc) {
        [fc takeoffWithCompletion:^(NSError * _Nullable error) {
            if (error) {
                ShowResult(@"Takeoff Error:%@", error.localizedDescription);
            }
            else
            {
                ShowResult(@"Takeoff Succeeded.");
            }
        }];
    }
    else
    {
        ShowResult(@"Component Not Exist");
    }
}

- (IBAction)onGoHomeButtonClicked:(id)sender {
    DJIFlightController* fc = [DemoComponentHelper fetchFlightController];
    if (fc) {
        [fc goHomeWithCompletion:^(NSError * _Nullable error) {
            if (error) {
                ShowResult(@"GoHome Error:%@", error.localizedDescription);
            }
            else
            {
                ShowResult(@"GoHome Succeeded.");
            }
        }];
    }
    else
    {
        ShowResult(@"Component Not Exist");
    }
}

- (IBAction)onLandButtonClicked:(id)sender {
    DJIFlightController* fc = [DemoComponentHelper fetchFlightController];
    if (fc) {
        [fc autoLandingWithCompletion:^(NSError * _Nullable error) {
            if (error) {
                ShowResult(@"Land Error:%@", error.localizedDescription);
            }
            else
            {
                ShowResult(@"Land Succeeded.");
            }
        }];
    }
    else
    {
        ShowResult(@"Component Not Exist");
    }
}
@end

File: Sample Code/SwiftSampleCode/DJISDKSwiftDemo/Components/FlightController/General/FCGeneralControlViewController.swift
//
//  FCGeneralControlViewController.swift
//  DJISDKSwiftDemo
//
//  Created by DJI on 16/1/6.
//  Copyright © 2016 DJI. All rights reserved.
//

import DJISDK
class FCGeneralControlViewController: DJIBaseViewController {
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view from its nib.
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    @IBAction func onTakeoffButtonClicked(_ sender: AnyObject) {
        let fc: DJIFlightController? = self.fetchFlightController()
        if fc != nil {
            fc!.takeoff(completion: {[weak self](error: Error?) -> Void in
                if error != nil {
                    self?.showAlertResult("Takeoff Error: \(error!.localizedDescription)")
                }
                else {
                    self?.showAlertResult("Takeoff Succeeded.")
                }
            })
        }
        else {
            self.showAlertResult("Component Not Exist")
        }
    }
    
    @IBAction func onGoHomeButtonClicked(_ sender: AnyObject) {
        let fc: DJIFlightController? = self.fetchFlightController()
        if fc != nil {
            fc!.goHome(completion: {[weak self](error: Error?) -> Void in
                if error != nil {
                    self?.showAlertResult("GoHome Error: \(error!.localizedDescription)")
                }
                else {
                    self?.showAlertResult("GoHome Succeeded.")
                }
            })
        }
        else {
            self.showAlertResult("Component Not Exist")
        }
    }
    
    @IBAction func onLandButtonClicked(_ sender: AnyObject) {
        let fc: DJIFlightController? = self.fetchFlightController()
        if fc != nil {
            fc!.autoLanding(completion: {[weak self](error: Error?) -> Void in
                if error != nil {
                    self?.showAlertResult("Land Error:\(error!.localizedDescription)")
                }
                else {
                    self?.showAlertResult("Land Succeeded.")
                }
            })
        }
        else {
            self.showAlertResult("Component Not Exist")
        }
    }
}
